EpiAware/src/EpiInfModels/InfectionODEProcess.jl

```diff
@@ -0,0 +1,182 @@
+"""
+A structure to hold the initial condition and parameters for an ODE (Ordinary Differential Equation) process.
+`params::ODEParams` is used in the method `generate_latent_infs(epi_model::InfectionODEProcess, params::ODEParams)`
+
+# Constructors
+- `ODEParams(; u0::VecOrMat, p::VecOrMat)`: Create an `ODEParams` object with the initial condition(s) `u0` and parameters `p`.
+
+# Fields
+- `u0::VecOrMat{T}`: The initial condition(s) for the ODE, which can be a vector or matrix of type `T`.
+- `p::VecOrMat{T}`: The parameters for the ODE, which can be a vector or matrix of type `T`.
+
+# Type Parameters
+- `T`: The type of the elements in `u0` and `p`.
+
+# Example
+```jldoctest
+using EpiAware
+params = ODEParams(; u0 = ones(10), p = [2, 3])
+
+# output
+
+ODEParams{Float64}([1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0], [2.0, 3.0])
+```
+"""
+struct ODEParams{T}
+    u0::VecOrMat{T}
+    p::VecOrMat{T}
+end
+
+function ODEParams(; u0::VecOrMat, p::VecOrMat)
+    T = promote_type(eltype(u0), eltype(p))
+    return ODEParams{T}(T.(u0), T.(p))
+end
+
+"""
+A structure representing an infection process modeled by an Ordinary Differential Equation (ODE).
+
+# Background
+
+The purpose of this structure is to define the behaviour of modelling an infection process
+using an ODE. We use the [`SciML` ecosystem](https://sciml.ai/) to define and solve the ODE.
+For `InfectionODEProcess` structs we focus on defining from a restricted set of ODE problems:
+- The initial condition `u0` must be a vector or matrix.
+- The parameters `p` must be a vector or matrix.
+- The output of the ODE should be interpreted as the infection incidence at each time point in
+`ts` via the function `sol2infs` which maps the solution object `sol` of the ODE solve to infection counts.
+
+# Fields
+- `prob::P`: The ODE problem instance, where `P` is a subtype of `ODEProblem`.
+- `ts::Vector{T}`: A vector of time points, where `T` is the type of the time points.
+- `solver::S`: The solver used for the ODE problem.
+- `sol2infs::F`: A function that maps the solution object of the ODE to infection counts.
+
+# Constructors
+- `InfectionODEProcess(prob::ODEProblem; ts, solver, sol2infs)`: Create an `InfectionODEProcess`
+object with the ODE problem `prob`, time points `ts`, solver `solver`, and function `sol2infs`.
+
+# Type Parameters
```

Comment:
We don't have this doc anywhere else (LLM spam?) do we need it?

Reply:
No we don't need it. This is indeed uncaught LLM spam